### PR TITLE
Star6

### PR DESCRIPTION
**Pull request by perevbnlgov**
In StMessage.h ostrstream replaced tp std::St_ostrstreamm, see Stsstreamm.h comments.
In Stsstream.h ostrstream was replaced by std::St_ostrstream to avoyd clash with real ç
and #define std::St_ostrstream ostrstream, to avoid clash with the real ostrstreamm

New class which support Cling macro call. In Root5 when Cint called macro from macro
these maccro are called independently. When submacro is defined but not called even if
such submacro does not exist or could not be called Cint is working smoothly.
In Cling, all the submacros are compiled even if this submacro is not really called.
In result your macro does not work if you have any problem in subbmacro, which is not even called.
To solve this preoblem using gROOT->ProcessLine(...) is proposed.
It is easy to do if there is no parameters. But parameters, especially with pointers is much
complicated. You must convert this pointers to the addresses.In result this process line
becames extremely long and unreadable. Example:

In Cint you need:
.x subbmacro(name1,name2);
TString Line(".x submacro(";

In Cling:
TString ts(".x subacro(");
ts += TString::ULLtoa((ULong64_t)name1,16); ts +=",";
ts += TString::ULLtoa((ULong64_t)name2,16); ts +=",";
ts+=")"
gROOT->ProcessLine("subMacro(char*)0x1234567,char*)0x1234567, ....")

Using method St::call it becames mor easy:
St::call(".x subMacro",name,name2)

### CODE REVIEW

**Comment by veprbl** [comment]
This is a Pull Request to merge perevbnlgov/Star6 branch into star-bnl/Star6. As far as I understand, Victor, these are both your branches, and nobody else is depending on those. Are you looking to get a review on some intermediate progress? Because, if not, I suppose you can just push directly to the Star6 branch of the main repo.

**Comment by perevbnlgov** [comment]
Yes, it is intermediate progress.
Victor

File: StarEtc/updRDict.csh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+##  	Libraies
+  set pcm = (`find $d -name '*.so'`)
+  foreach l ($pcm)
+    set o = ${l:t}
+    set o = ${STAR_LIB}/${o}
+    if (-e "$o") continue
+    rm -f $o
+    ln -s ${STAR_OBJ}/$l $o
+  end
+end
```
Comment (plexoos) [inline StarEtc/updRDict.csh]: Please see https://github.com/star-bnl/star-sw/pull/121
The main points to take away from #121 are:

- rootcint in ROOT6 (a.k.a rootcling) apparently produces .pcm files without any additional options
- The produced .pcm files are installed in $STAR_LIB via the Install directive

**Comment by perevbnlgov** [comment]
>  	* The produced .pcm files are installed in $STAR_LIB via the Install
> directive
Could you please show me to install .pcm to STAR_LIB
Victor

**Comment by plexoos** [comment]
> Could you please show me to install .pcm to STAR_LIB

On the pull request page you can click on the "Files changed" tab and see the changes. Here is the direct link:
https://github.com/star-bnl/star-sw/pull/121/files

**Comment by perevbnlgov** [comment]
> see the changes. Here is the direct link:
> https://github.com/star-bnl/star-sw/pull/121/files

In this code I do not see not sylink , no ln -s.
ay be you use copy of .pcm file instead of link?
Victor

**Comment by plexoos** [comment]
> In this code I do not see not sylink , no ln -s. ay be you use copy of .pcm file instead of link?

Symlinking is done by this Cons command `Install $env somedir, somefile`
It is similar to how shared libraries are installed

**Comment by perevbnlgov** [comment]
Ok, is it already in Star6 branch in the main repository? Could I try 
it?
Victor

**Comment by plexoos** [comment]
Yes, these changes are on the "main" branch of the primary repository "star-bnl/star-sw" now (just merged). I would suggest you do the following:

1. Verify that your "origin" remote is set to the primary repository. You should see something like this:
```
$ git remote -v
origin  git@github.com:star-bnl/star-sw.git (fetch)
origin  git@github.com:star-bnl/star-sw.git (push)
```

2. Then do:

```
git fetch origin
git checkout Star6
git rebase origin/main
git push -f origin
```

3. Now your "Star6" branch is on top of the "main" branch and includes all the changes

**Comment by perevbnlgov** [comment]
> Yes, these changes are on the "main" branch of the primary repository
> "star-bnl/star-sw" now (just merged). I would suggest you do the
Does that mean, that I can compile Star6 using mgr/ from the main 
branch?

Victor

**Comment by plexoos** [comment]
> Does that mean, that I can compile Star6 using mgr/ from the main branch?

Yes, when you rebase `Star6` onto `main` all of your files will have changes from the two branches. To illustrate: 

output

From [here](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#conceptual-overview)

**Comment by perevbnlgov** [comment]
I tried to compile my Star6 by mgr from the main.
StarVMC/StarGeometry failed
StTable failed during loading library (compilation OK)
But *.pcm file linked to StarLib correctly

Victor

**Comment by plexoos** [comment]
> I tried to compile my Star6 by mgr from the main.

Quite honestly, I don't even know what you mean by this.

> StarVMC/StarGeometry failed
> StTable failed during loading library (compilation OK)

If you'd like someone to follow up on this, you should create an issue (a.k.a. ticket) with how to reproduce.
But keep in mind that the code on the main branch does compile without issues and the problem can be in the changes you introduce on top of that

> But *.pcm file linked to StarLib correctly

Good.

File: StarVMC/xgeometry/xgeometry.age [hunk 2]
```diff
@@ -1,4625 +0,0 @@
-* $Id: xgeometry.age,v 1.43 2021/02/25 15:48:34 jwebb Exp $
```
Comment (klendathu2k) [inline StarVMC/xgeometry/xgeometry.age]: I assume this deletion is a mistake.  xgeometry.age _steers the construction of the geometry for starsim_.  Please restore.

**Review by klendathu2k — CHANGES_REQUESTED** [review]

**Pull request closed without merging**